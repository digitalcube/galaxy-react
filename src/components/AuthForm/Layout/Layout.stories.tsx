import React, { FC, useEffect, useState } from 'react';
import { Meta, Story } from '@storybook/react';
import { AuthFormLayout, AuthFormLayoutProps } from './Layout';
import { AuthFormHeader } from '../../AuthForm';
import { Button } from '../../Buttons';
import { ImageShifterLogo } from '../../Images';

type AuthFormLayoutExampleProps = {
  status: '' | 'failure' | 'inprogress';
} & Pick<AuthFormLayoutProps, 'inprogress' | 'success'>;
const AuthFormLayoutExample: FC<AuthFormLayoutExampleProps> = ({
  status,
  inprogress,
  success,
}) => {
  const [currentStatus, setCurrentStatus] = useState(status);
  useEffect(() => {
    setCurrentStatus(status);
  }, [status, setCurrentStatus]);
  return (
    <AuthFormLayout
      variant="login"
      status={currentStatus}
      inprogress={inprogress}
      success={success}
    >
      <form
        noValidate
        onSubmit={(e) => {
          e.preventDefault();
        }}
      >
        <AuthFormHeader
          logo={<ImageShifterLogo width="36" height="46" alt="Shifter" />}
          title="Log In"
          variant="login"
        />
        <Button type="submit" className="mt-4" fontWeight="bold" block disabled>
          Log In
        </Button>
      </form>
    </AuthFormLayout>
  );
};

const meta: Meta = {
  title: 'Components/AuthForm/Layouts',
  component: AuthFormLayoutExample,
  argTypes: {
    status: {
      options: ['default', 'inprogress', 'failure', 'success'],
      control: { type: 'radio' },
    },
  },
  parameters: {
    controls: {
      expanded: true,
    },
  },
};

export default meta;

const Template: Story<AuthFormLayoutExampleProps> = (args) => (
  <AuthFormLayoutExample {...args} />
);

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});
Default.args = {
  status: '',
  inprogress: {
    title: '',
    message: '',
  },
  success: {
    title: '',
    message: '',
  },
};
