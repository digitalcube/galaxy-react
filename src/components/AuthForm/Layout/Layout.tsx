import React, { FC, PropsWithChildren, ReactNode } from 'react';
import { FaCheck } from 'react-icons/fa';

export type AuthFormLayoutProps = PropsWithChildren<{
  status: '' | 'inprogress' | 'failure' | 'success';
  inprogress?: {
      title?: string;
      message?: ReactNode;
  }
  success?: {
    title?: string;
    message?: ReactNode
  }
}>;

export const AuthFormLayout: FC<AuthFormLayoutProps> = ({
  status,
  children,
  inprogress,
  success,
}) => {
  const classNames = [
    'login',
    'px-4',
    'py-4',
    'rounded',
    'd-flex',
    'flex-column',
    'justify-content-center',
  ];
  if (['inprogress', 'success'].includes(status)) classNames.push('success');
  if (status === 'failure') classNames.push('error');

  const inprogressMessageTitle = inprogress?.title || null
  const inprogressMessage = inprogress?.message || null
  const succeededMessageTitle = success?.title || null
  const succeededMessage = success?.message || null
  return (
    <div className={classNames.join(' ')}>
      {status === 'inprogress' ? (
        <div className="mb-4 pb-2 login-header">
          <div className="mx-auto mb-3 pt-2 d-flex align-items-center justify-content-center rounded-circle login-success-icon">
            <i aria-hidden="true">
              <FaCheck className="d-block" />
            </i>
          </div>
          <h1 className="mb-0 font-weight-bold text-center logging-text">
            {inprogressMessageTitle || 'Processing...'}
          </h1>
          {inprogressMessage ? (
            <div className="mt-4 font-weight-bold text-center send-email">
              {inprogressMessage}
            </div>
          ) : null}
        </div>
      ) : null}
      {status === 'success' ? (

          <div className="mb-4 pb-2 login-header">
            <div className="mx-auto mb-3 pt-2 d-flex align-items-center justify-content-center rounded-circle login-success-icon">
              <i aria-hidden="true">
                <FaCheck className="d-block" />
              </i>
            </div>
            <h1 className="mb-0 font-weight-bold text-center logging-text">
              {succeededMessageTitle || 'Complete'}
            </h1>
            {inprogressMessage ? (
              <div className="mt-4 font-weight-bold text-center send-email">
                {succeededMessage}
              </div>
            ) : null}
          </div>

      ): null}
      {!['inprogress', 'success'].includes(status) ? children: null}
    </div>
  );
};
