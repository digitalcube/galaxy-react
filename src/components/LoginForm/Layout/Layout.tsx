import React, { FC, PropsWithChildren } from 'react';
import { FaCheck } from 'react-icons/fa';

export type LoginFormLayoutProps = PropsWithChildren<{
  status: '' | 'inprogress' | 'failure';
  inprogressMessageTitle?: string;
  inprogressMessage?: string;
}>;

export const LoginFormLayout: FC<LoginFormLayoutProps> = ({
  status,
  children,
  inprogressMessageTitle,
  inprogressMessage,
}) => {
  const classNames = [
    'login',
    'px-4',
    'py-4',
    'rounded',
    'd-flex',
    'flex-column',
    'justify-content-center',
  ];
  if (status === 'inprogress') classNames.push('success');
  if (status === 'failure') classNames.push('error');
  return (
    <div className={classNames.join(' ')}>
      {status === 'inprogress' ? (
        <div className="mb-4 pb-2 login-header">
          <div className="mx-auto mb-3 pt-2 d-flex align-items-center justify-content-center rounded-circle login-success-icon">
            <i aria-hidden="true">
              <FaCheck className="d-block" />
            </i>
          </div>
          <h1 className="mb-0 font-weight-bold text-center logging-text">
            {inprogressMessageTitle || 'Logging you in...'}
          </h1>
          {inprogressMessage ? (
            <div className="mt-4 font-weight-bold text-center send-email">
              {inprogressMessage}
            </div>
          ) : null}
        </div>
      ) : (
        children
      )}
    </div>
  );
};
