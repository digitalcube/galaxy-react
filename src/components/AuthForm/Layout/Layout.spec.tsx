import { render } from '@testing-library/react';
import React from 'react';
import { Default as Layout } from './Layout.stories';

describe('Layout', () => {
  it('to match snapshot', () => {
    const { asFragment } = render(
      <Layout {...(Layout.args as any)} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
