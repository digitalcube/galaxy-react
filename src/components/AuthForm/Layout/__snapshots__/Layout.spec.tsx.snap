// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="login px-4 py-4 rounded d-flex flex-column justify-content-center mx-3"
  >
    <form
      novalidate=""
    >
      <div
        class="mb-4 pb-2 login-header"
      >
        <div
          class="mb-4 text-center login-icon"
        >
          <img
            alt="Shifter"
            height="46"
            width="36"
          />
        </div>
        <h1
          class="mb-0 font-weight-bold text-center login-title"
        >
          Log In
        </h1>
      </div>
      <button
        class="mt-4 btn-colored font-weight-bold undefined btn btn-secondary btn-block disabled"
        disabled=""
        type="submit"
      >
        Log In
      </button>
    </form>
  </div>
</DocumentFragment>
`;
